pkg/controller/application/application_controller.go

```diff
@@ -215,6 +235,17 @@ func (c *Controller) enqueueApp(obj interface{}) {
 	c.appWorkqueue.Add(key)
 }
 
+func (c *Controller) enqueueAppAfterDuration(obj interface{}) {
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		runtime.HandleError(err)
+		return
+	}
+
+	duration := shippercontroller.GetDuration(c.resyncPeriod)
```

Comment:
This is the duration I meant above: if it's going to stay like this for a longer period of time, we can easily ignore same-UUID notifications. Another danger I see here is: it's an ever-growing queue which can accumulate itself within shipper runtime and therefore cause resource leakage.